### PR TITLE
Fix dep warnings from pandas and mpl

### PR DESCRIPTION
**Pull request by nabobalis**

### CODE REVIEW

File: sunpy/map/compositemap.py [hunk 1]
```diff
@@ -466,7 +466,8 @@ def plot(self, axes=None, annotate=True,
                 ret.append(m.draw_contours(m.levels, **params))
 
                 # Set the label of the first line so a legend can be created
-                ret[-1].collections[0].set_label(m.name)
+                ret[-1].set_label(m.name)
+                #ret[-1].collections[0].set_label(m.name)
```
Comment (nabobalis) [inline sunpy/map/compositemap.py:470]: WIP

File: sunpy/map/compositemap.py [hunk 2]
```diff
@@ -2,18 +2,22 @@
...
 
 import astropy.units as u
+from astropy.utils.introspection import minversion
 
 from sunpy.map import GenericMap
 from sunpy.util import expand_list, get_keywords, get_set_methods
 from sunpy.util.decorators import add_common_docstring
 from sunpy.visualization import axis_labels_from_ctype, peek_show, wcsaxes_compat
 
+MATPLOTLIB_LT_3_8 = not minversion(matplotlib, "3.8.dev")
```
Comment (nabobalis) [inline sunpy/map/compositemap.py:19]: Stolen with love from Astropy https://github.com/astropy/astropy/pull/15004/files

**Comment by nabobalis** [comment]
devdeps is now:

```
FAILED ../../.tox/py311-devdeps/lib/python3.11/site-packages/sunpy/io/special/asdf/tests/test_coordinate_frames.py::test_saveframe[HeliographicStonyhurst]
FAILED ../../.tox/py311-devdeps/lib/python3.11/site-packages/sunpy/io/special/asdf/tests/test_coordinate_frames.py::test_saveframe[HeliographicCarrington]
FAILED ../../.tox/py311-devdeps/lib/python3.11/site-packages/sunpy/io/special/asdf/tests/test_coordinate_frames.py::test_saveframe[Heliocentric]
FAILED ../../.tox/py311-devdeps/lib/python3.11/site-packages/sunpy/io/special/asdf/tests/test_coordinate_frames.py::test_saveframe[Helioprojective]
FAILED ../../.tox/py311-devdeps/lib/python3.11/site-packages/sunpy/io/special/asdf/tests/test_coordinate_frames.py::test_saveframe[HeliocentricEarthEcliptic]
FAILED ../../.tox/py311-devdeps/lib/python3.11/site-packages/sunpy/io/special/asdf/tests/test_coordinate_frames.py::test_saveframe[GeocentricSolarEcliptic]
FAILED ../../.tox/py311-devdeps/lib/python3.11/site-packages/sunpy/io/special/asdf/tests/test_coordinate_frames.py::test_saveframe[HeliocentricInertial]
FAILED ../../.tox/py311-devdeps/lib/python3.11/site-packages/sunpy/io/special/asdf/tests/test_coordinate_frames.py::test_saveframe[GeocentricEarthEquatorial]
FAILED ../../.tox/py311-devdeps/lib/python3.11/site-packages/sunpy/io/special/asdf/tests/test_coordinate_frames.py::test_saveframe_arr[HeliographicStonyhurst]
FAILED ../../.tox/py311-devdeps/lib/python3.11/site-packages/sunpy/io/special/asdf/tests/test_coordinate_frames.py::test_saveframe_arr[HeliographicCarrington]
FAILED ../../.tox/py311-devdeps/lib/python3.11/site-packages/sunpy/io/special/asdf/tests/test_coordinate_frames.py::test_saveframe_arr[Heliocentric]
FAILED ../../.tox/py311-devdeps/lib/python3.11/site-packages/sunpy/io/special/asdf/tests/test_coordinate_frames.py::test_saveframe_arr[Helioprojective]
FAILED ../../.tox/py311-devdeps/lib/python3.11/site-packages/sunpy/io/special/asdf/tests/test_coordinate_frames.py::test_saveframe_arr[HeliocentricEarthEcliptic]
FAILED ../../.tox/py311-devdeps/lib/python3.11/site-packages/sunpy/io/special/asdf/tests/test_coordinate_frames.py::test_saveframe_arr[GeocentricSolarEcliptic]
FAILED ../../.tox/py311-devdeps/lib/python3.11/site-packages/sunpy/io/special/asdf/tests/test_coordinate_frames.py::test_saveframe_arr[HeliocentricInertial]
FAILED ../../.tox/py311-devdeps/lib/python3.11/site-packages/sunpy/io/special/asdf/tests/test_coordinate_frames.py::test_saveframe_arr[GeocentricEarthEquatorial]
FAILED ../../.tox/py311-devdeps/lib/python3.11/site-packages/sunpy/io/special/asdf/tests/test_coordinate_frames.py::test_hpc_observer_version
FAILED ../../.tox/py311-devdeps/lib/python3.11/site-packages/sunpy/io/special/asdf/tests/test_coordinate_frames.py::test_hcc_observer_version
```

File: sunpy/map/compositemap.py (hunk 2 above)
Comment (wtbarnes) [reply sunpy/map/compositemap.py:19]: Remove the `not` and invert your conditional below?

Comment (nabobalis) [reply sunpy/map/compositemap.py:19]: Sure.

File: sunpy/map/compositemap.py [hunk 3]
```diff
@@ -466,7 +470,11 @@ def plot(self, axes=None, annotate=True,
                 ret.append(m.draw_contours(m.levels, **params))
 
                 # Set the label of the first line so a legend can be created
-                ret[-1].collections[0].set_label(m.name)
+                # TODO: remove when we depend on matplotlib 3.8 or later
```
Comment (wtbarnes) [inline sunpy/map/compositemap.py:473]: Let's try to keep track of this, maybe via an issue or something.

**Review by wtbarnes — APPROVED** [review]

File: sunpy/map/compositemap.py (hunk 3 above)
Comment (nabobalis) [reply sunpy/map/compositemap.py:473]: I will open an issue!

**Pull request merged**